### PR TITLE
chore(rolldown): bump oxc to 0.31.0

### PR DESCRIPTION
**Pull request by Boshen**

### CODE REVIEW

File: crates/rolldown/tests/esbuild/dce/remove_unused_pure_comment_calls/artifacts.snap [hunk 1]
```diff
@@ -8,7 +8,7 @@ source: crates/rolldown_testing/src/integration_test.rs
 ```js
 
 //#region entry.js
-function bar() {}
+// @__PURE__ // @__PURE__ // @__PURE__ // @__PURE__ // #__PURE__ // #__PURE__ // #__PURE__ // #__PURE__ function bar() {}
```
Comment (Boshen) [inline crates/rolldown/tests/esbuild/dce/remove_unused_pure_comment_calls/artifacts.snap]: Oh no!

File: crates/rolldown/tests/rolldown/misc/common_js_min/artifacts.snap [hunk 2]
```diff
@@ -6,5 +6,5 @@ source: crates/rolldown_testing/src/integration_test.rs
...
+var __create=Object.create;var __defProp=Object.defineProperty;var __getOwnPropDesc=Object.getOwnPropertyDescriptor;var __getOwnPropNames=Object.getOwnPropertyNames;var __getProtoOf=Object.getPrototypeOf;var __hasOwnProp=Object.prototype.hasOwnProperty;var __commonJSMin=(l,m)=>()=>(m||l((m={exports:{}}).exports,m),m.exports);var __copyProps=(l,m,n,o)=>{if(m&&typeof m==="object"||typeof m==="function"){for(var p=__getOwnPropNames(m),q=0,r=p.length,s;q<r;q++)if(s=p[q],!__hasOwnProp.call(l,s)&&s!==n)__defProp(l,s,{get:((t)=>m[t]).bind(null,s),enumerable:!(o=__getOwnPropDesc(m,s))||o.enumerable})}return l};var __toESM=(l,m,n)=>(n=l!=null?__create(__getProtoOf(l)):{},__copyProps(m||!l||!l.__esModule?__defProp(n,"default",{value:l,enumerable:!0}):n,l));var require_shared=__commonJSMin((exports,m)=>{m.exports=123});var import_shared=__toESM(require_shared());assert.equal(import_shared.foo,123);
```
Comment (Boshen) [inline crates/rolldown/tests/rolldown/misc/common_js_min/artifacts.snap:9]: Regression.

File: packages/rolldown/src/binding.d.ts [hunk 3]
```diff
@@ -627,23 +633,13 @@ export declare function transform(filename: string, sourceText: string, options?
  */
 export interface TransformOptions {
   sourceType?: 'script' | 'module' | 'unambiguous' | undefined
+  /** Treat the source text as `js`, `jsx`, `ts`, or `tsx`. */
+  lang?: 'js' | 'jsx' | 'ts' | 'tsx'
```
Comment (Boshen) [inline packages/rolldown/src/binding.d.ts:643]: cc @underfin

File: crates/rolldown/tests/esbuild/dce/remove_unused_pure_comment_calls/artifacts.snap (hunk 1 above)
Comment (Dunqing) [reply crates/rolldown/tests/esbuild/dce/remove_unused_pure_comment_calls/artifacts.snap]: Let me follow this, I think I know what happened here.

File: packages/rolldown/src/binding.d.ts (hunk 3 above)
Comment (underfin) [reply packages/rolldown/src/binding.d.ts:643]: OK. Get it.

File: crates/rolldown/tests/rolldown/misc/common_js_min/artifacts.snap (hunk 2 above)
Comment (Boshen) [reply crates/rolldown/tests/rolldown/misc/common_js_min/artifacts.snap:9]: Which minifier is used here? I'm confused.

**Review by IWANABETHATGUY — APPROVED** [review]

**Pull request merged**